pkg/ingester/ingester_test.go

```diff
@@ -6108,6 +6108,113 @@ func Test_Ingester_OutOfOrder(t *testing.T) {
 	assert.Equal(t, int64(30*60), usagestats.GetInt(maxOutOfOrderTimeWindowSecondsStatName).Value())
 }
 
+// Test_Ingester_OutOfOrder_CompactHead tests that the OOO head is compacted
+// when the compaction is forced or when the TSDB is idle.
+func Test_Ingester_OutOfOrder_CompactHead(t *testing.T) {
+	cfg := defaultIngesterTestConfig(t)
+	cfg.BlocksStorageConfig.TSDB.HeadCompactionInterval = 1 * time.Hour      // Long enough to not be reached during the test.
+	cfg.BlocksStorageConfig.TSDB.HeadCompactionIdleTimeout = 1 * time.Second // Testing this.
+	cfg.TSDBConfigUpdatePeriod = 1 * time.Second
+
+	l := defaultLimitsTestConfig()
+	tenantOverride := new(TenantLimitsMock)
+	tenantOverride.On("ByUserID", userID).Return(nil)
```

Comment:
[nit] would be better to use `validation.NewMockTenantLimits(limits)` and update `limits` later in the test, rather than string constants to mock function calls.